.github/workflows/publish-package.yml

```diff
@@ -10,22 +10,22 @@ on:
 jobs:
   deploy:
     runs-on: ubuntu-latest
+    environment: pypi
+    permissions:
+      id-token: write
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@ee0669bd1cc54295c223e0bb666b733df41de1c5 # pin@v2.7.0
```

Comment:
@MSWechsler curious about this. Can we not just pin to a version over a hash? Might make this more readable? Same would go for all the other PRs and instance of this.